src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureCredentialsProvider.java

```diff
@@ -189,6 +190,15 @@ private static Collection<IdCredentials> fetchCredentials() {
                             credentials.add(cred);
                             break;
                         }
+                        case "secretFile": {
+                            String fileName = tags.get("fileName");
+                            if(fileName.isEmpty()){
+                                fileName = "defaultFileName.txt";
```

Comment:
I think this could be better as just the secret name?

I was just looking at how regular credentials do it and I think its done by the file name used when uploaded. I think the secret key name would be the closest to that.

You could then just use `.txt` as the default or check the content-type attribute perhaps.

Reply:
changed to use the secret name in the vault as default name with .txt as default extension 

the user can always set an explicit name with the tag ( to not make the a complicated logic to guess what the file extension should be :) )